src/batch-client/index.ts

```diff
@@ -169,6 +169,49 @@ function normalizeMessage(
 	);
 }
 
+const hasUnreadDescendent = (folder: any) => {
+	if (
+		folder[
+			folder.name === 'Drafts' || folder.name === 'Outbox'
+				? 'nonFolderItemCount'
+				: 'unread'
+		] > 0
+	) {
+		return true;
+	}
+
+	const folderArray = get(folder, 'folders') || [];
+	for (let i = 0; i < folderArray.length; i++) {
+		if (hasUnreadDescendent(folderArray[i])) {
+			return true;
+		}
+	}
+
+	return false;
+};
+
+const setUnreadDescendentFlag = (folder: any) => {
+	const folderArray = get(folder, 'folders') || [];
+	const view =  get(folder, 'view');
+
+	// setting this flag only in message view has we dont want to show unread count in
+	// other views
+	if (!view || view === FolderView.Message) {
+		folder = {
+			...folder,
+			unreadDescendent: hasUnreadDescendent(folder)
+		};
+
+		if (folderArray) {
+			folder = {
+				...folder,
+				folders: folderArray.map(setUnreadDescendentFlag)
```

Comment:
hasUnreadDescendent is already doing a loop on subfolders, here we are also doing same.
can we reduce looping required to achieve end result?

Reply:
Have tried,but couldnt find a solution ,to reduce looping , for certian level when a unreadDescendent is laready present it wont loop over , but we do need to check into deeper level 
https://github.com/Zimbra/zm-api-js-client/pull/447/files#diff-2f62190d7ad815da92e5dcf270fdd6e290f3362cf453d2951e2f4d7ed46e2badR182-R183
t